### PR TITLE
[Fix] Notify that user cannot send a message because one of the devices doesn't support LH

### PR DESCRIPTION
**Pull request by KaterinaWire**
### Issues
When a user who is under LH has one of the devices that doesn't support LH (not updated client), there's an 412 error from BE and the user is unable to send a message.

### Causes
All clients are not updated.

### Solutions
Handle the `403` error with `"missing-legalhold-consent"` label and notify the user.

## Notes
https://wearezeta.atlassian.net/browse/SQSERVICES-526

### CODE REVIEW

File: Sources/Request Strategies/Client Message/ClientMessageTranscoder.swift [hunk 1]
```diff
@@ -230,6 +230,16 @@ extension ClientMessageTranscoder {
...
         }
+
+        if response.httpStatus == 412,
+           let label = response.payloadLabel(), label == "missing-legalhold-consent" {
+            managedObjectContext.zm_userInterface.performGroupedBlock { [weak self] in
+                guard let `self` = self else { return }
+                NotificationInContext(name: ZMConversation.failedToSendMessageNotificationName,
+                                      context: self.managedObjectContext.notificationContext).post()
+            }
+        }
```
Comment (johnxnguyen) [inline Sources/Request Strategies/Client Message/ClientMessageTranscoder.swift:241]: Suggestion to clean this up a little: 

```swift
if response.httpStatus == 412 && response.payloadLabel() == "missing-legalhold-consent" {
  managedObjectContext.zm_userInterface.performGroupedBlock { [weak self] in
    guard let context = self?.managedObjectContext.notificationContext else { return }
    NotificationInContext(name: ZMConversation.failedToSendMessageNotificationName, context: context).post()
  }
}
```

**Review by johnxnguyen — CHANGES_REQUESTED** [review]
It would be good to add a test that the notification is sent.

**Comment by typfel** [comment]
Is there any specs for how this should behave? I'm asking because the `ClientMessageTranscoder` is not the only place where you can receive this error.

ClientMessageTranscoder
AssetClientMessageRequestStrategy
LinkPreviewUploadRequestStrategy
GenericMessageRequestStrategy which is used in CallingRequestStrategy, AvailabilityRequestStrategy, DeliveryReceiptRequestStrategy

All send encrypted messages which can then in theory receive this error.

I guess mostly when sense show the alert when the user initiates the message sending and not when a delivery receipt is sent but that's up for discussion I guess.

**Comment by KaterinaWire** [comment]
> LinkPreviewUploadRequestStrategy

We have only this ticket https://wearezeta.atlassian.net/browse/SQSERVICES-524. 
I agree that it should cover sending not only a text message, but assets as well.

About the `LinkPreviewUploadRequestStrategy`, we won't trigger it as the message will not be sent. Correct?
The calling is not a part of this task, but I will add this question to the parent ticket, because it's relevant to other platforms.

So, as I understand it, only 2 strategies (ClientMessageTranscoder, AssetClientMessageRequestStrategy) are required to handle this error. @typfel  Are you agree?

File: Sources/Request Strategies/Client Message/ClientMessageTranscoder.swift (hunk 1 above)
Comment (KaterinaWire) [reply Sources/Request Strategies/Client Message/ClientMessageTranscoder.swift]: Done

**Review by johnxnguyen — APPROVED** [review]

**Review by typfel — APPROVED** [review]

**Comment by KaterinaWire** [comment]
@zenkins release minor

**Pull request merged**